From 8dcab304eb110f7546719d0f0861ef6599f575e1 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:54:23 -0400
Subject: [fs] ext4: remove an unneeded check in mext_page_mkuptodate()

Message-id: <1414058125-4183-110-git-send-email-lczerner@redhat.com>
Patchwork-id: 98193
O-Subject: [RHEL 7.1 PATCH v2 109/171] ext4: remove an unneeded check in mext_page_mkuptodate()
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Dan Carpenter <dan.carpenter@oracle.com>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit df3a98b0865467ae8033c55ebb514debd69b4e59

 "err" is zero here, there is no need to check again.

 Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/move_extent.c b/fs/ext4/move_extent.c
index 773b503..f39a88a 100644
--- a/fs/ext4/move_extent.c
+++ b/fs/ext4/move_extent.c
@@ -861,8 +861,7 @@ mext_page_mkuptodate(struct page *page, unsigned from, unsigned to)
    }
    if (!buffer_mapped(bh)) {
     zero_user(page, block_start, blocksize);
-    if (!err)
-     set_buffer_uptodate(bh);
+    set_buffer_uptodate(bh);
     continue;
    }
   }
-- 
1.7.1